scipy/optimize/_linprog_util.py

```diff
@@ -11,6 +11,65 @@
     )
 
 
+class LPProblem(dict):
+    """ Represents a linear-programming problem.
+
+    Attributes
+    ----------
+    c : 1D array
+        The coefficients of the linear objective function to be minimized.
+    A_ub : 2D array, optional
+        The inequality constraint matrix. Each row of ``A_ub`` specifies the
+        coefficients of a linear inequality constraint on ``x``.
+    b_ub : 1D array, optional
+        The inequality constraint vector. Each element represents an
+        upper bound on the corresponding value of ``A_ub @ x``.
+    A_eq : 2D array, optional
+        The equality constraint matrix. Each row of ``A_eq`` specifies the
+        coefficients of a linear equality constraint on ``x``.
+    b_eq : 1D array, optional
+        The equality constraint vector. Each element of ``A_eq @ x`` must equal
+        the corresponding element of ``b_eq``.
+    bounds : sequence, optional
+        A sequence of ``(min, max)`` pairs for each element in ``x``, defining
+        the minimum and maximum values of that decision variable. Use ``None`` to
+        indicate that there is no bound. By default, bounds are ``(0, None)``
+        (all decision variables are non-negative).
+        If a single tuple ``(min, max)`` is provided, then ``min`` and
+        ``max`` will serve as bounds for all decision variables.
+    x0 : 1D array, optional
+        Guess values of the decision variables, which will be refined by
+        the optimization algorithm. This argument is currently used only by the
+        'revised simplex' method, and can only be used if `x0` represents a
+        basic feasible solution.
+    """
+
+    def __init__(self, *args, **kwargs):
+        for key in ['A_ub', 'b_ub', 'A_eq', 'b_eq', 'bounds', 'x0']:
```

Comment:
`c`?

Reply:
`c` was left out intentionally as it's the only mandatory argument for an LP - this way if `c` is called without being initialized with a value first, it would throw an `AttributeError` :)
Let me know if this is not the expected behaviour xD

Reply:
Maybe accept `c` as a parameter (independent of `args`/`kwargs`) with no default value?

Reply:
Would this suffice?
```Python
def __init__(self, c, *args, **kwargs):
    for key in ['A_ub', 'b_ub', 'A_eq', 'b_eq', 'bounds', 'x0']:
        self[key] = None
    dict.__init__(self, *args, **dict(kwargs, c=c))
```
Nitpick: the difference in initialization (compared to a regular dict) might be worth documenting:
```Python
lp1 = LPProblem(c=[-1, 4], A_ub=[[-3, 1], [1, 2]], b_ub=[6, 4])  # probably the most used and readable approach
lp2 = LPProblem({'c': [-1, 4], 'A_ub': [[-3, 1], [1, 2]], 'b_ub': [6, 4]})  # unusable; lp2.c becomes the entire dictionary
lp3 = LPProblem([-1, 4], {'A_ub': [[-3, 1], [1, 2]], 'b_ub': [6, 4]})  # fine
lp4 = LPProblem([-1, 4], {'c': [2, 3], 'A_ub': [[-3, 1], [1, 2]], 'b_ub': [6, 4]})  # lp4.c = [-1, 4] not [2, 3]
```
I can add example usage in the comment string if we go with this approach :)

Reply:
`lp2`'s definition can work if you check whether the first parameter is a dictionary or not. If it is, you can assume it contains `c`, etc...
I don't think we should encourage definitions like `lp3` and `lp4`. 

Also, we didn't really discuss whether this class would be public or not, but it sounds like we're assuming it will be. I think that's fine, so users would then be able to define an `LPProblem` and use `linprog` to solve it:
"""
lp = LPProblem(...)
linprog(lp)
"""
but we will still need to allow the original syntax for calling `linprog`.

But I think @Kai-Striega's original purpose was to simplify passing all the information around _internally_.

Reply:
> The `lp2` initialization can work if you check whether the first parameter of `__init__` is a dictionary or not. If not, then the first argument is `c`. If it is, assume it contains the entire problem statement (including constraints, bounds, and guess).
I would advise against performing this check to avoid ambiguity in the variable `c` and type-checking in general. Instead I can add a note/ example usage in the docstring saying not to pass in a whole dictionary in a `lp2`-manner.

> I don't think we should encourage definitions like `lp3` and `lp4`.
Agreed! Was just trying to list out the corner cases :)

I think we should keep this class internal-use-only (and keep the parameter-passing for public methods as-is) to stick with the original motivation xD

Reply:
Another case, that currently works, is to specify values as arguments, e.g.:

```python
lp5 = LPProblem([-1, 4], [[-3, 1], [1, 2]], [6, 4])
```

If we do change publically this way would need to be doable.